preprocessing/data_explorer.py

```diff
@@ -147,3 +149,105 @@ def explore_data(dataframe):
+
+        while len(forest) < self.number_trees:
+            # Select random subsets of points uniformly from point set
+            ixs = np.random.choice(self.number_of_rows, size=sample_size_range,
+                                   replace=False)
+            # Add sampled trees to forest
+            trees = [rrcf.RCTree(self.data_array[ix], index_labels=ix)
+                     for ix in ixs]
+            forest.extend(trees)
+        return forest
+
+    def codisp_average(self, forest: list) -> pd.Series:
+        """ Average score
+
+        It calculates the score for each data point in all the trees and then calculates the average value of the score.
+
+        :param list forest: A list of the trees that are created randomly.
+        :return:
+                - pd.series avg_codisp: The averege score of each data point calculated from all the constructed trees
+        """
+
+        # Compute average CoDisp
+        avg_codisp = pd.Series(0.0, index=np.arange(self.number_of_rows))
+        index = np.zeros(self.number_of_rows)
+        for tree in forest:
+            codisp = pd.Series({leaf: tree.codisp(leaf)
+                                for leaf in tree.leaves})
+            avg_codisp[codisp.index] += codisp
+            np.add.at(index, codisp.index.values, 1)
+        avg_codisp /= index
+        return avg_codisp
+
+
+def outliers_detector(data_array: np.array,
+                      number_trees: int = 100,
+                      tree_size: int = 128,
+                      outliers_threshold: float = 0.99) -> Tuple[pd.Series, List]:
+    """ RRCF outliers detector
+
+    This functions detect if there is outliers in the given dataset. It uses the Robust Random Cut Forest Algorithm.
+    For more information about this method please check the following link\n
+    https://klabum.github.io/rrcf/random-cut-tree.html
+
+    :param np.array data_array: A numpy array which has the size of n x m. The missing values are replaced by the means
+                                and the elements are only numeric
+    :param int number_trees: The number of the trees that will be randomly generated to create the forest
+    :param int tree_size: The number of the elements that a tree should contain.
+    :param float outliers_threshold: The score' threshold which above it, the data point is considered as an outlier.
+    :return:
+            - pd.series avg_codisp: The averege score of each data point calculated from all the constructed trees
+            - list outliers_index: bool type list. If the value is true, then the data point is an outlier.
+    """
+
+    avg_codisp = []
+    outliers_index = []
+
+    try:
+        # Construct forest
+        detect_outliers = OutliersDetection(data_array, number_trees, tree_size)
+        forest = detect_outliers.construct_force()
+        avg_codisp = detect_outliers.codisp_average(forest)
+
+        outliers_index = avg_codisp > avg_codisp.quantile(outliers_threshold)
```

Comment:
Is this operation expected to yield a boolean? Or is the `>` operation implemented as a protocol inside the `codisp_average` data structure? The reason I am asking is that the function is documented to return a `Tuple[pd.Series, List]` which in this case would return a `Tuple[pd.Series, bool]`, if my assumption holds true. This might break code @`preprocessing/data_type_detector.py#290`, namely:

```python
_, is_outlier = outliers_detector(data_points)
if sum(is_outlier) > 0:            # <--- TypeError: 'bool' object is not iterable
    # ... etc
```